### ISSUE TITLE
Debug and Clone traits for struct Header

### ISSUE THREAD

**Issue by inashivb** [staktrace/mailparse#83]
Hello!

I have been trying to use the `Headers` struct of this crate as a part of another struct in my code (`MimeDecode`). I wish to print and clone `MimeDecode` but I get an error
```
 ^^^^^^^^^^^^^ `Headers<'_>` cannot be formatted using `{:?}` because it doesn't implement `Debug`
```
where
```
#[derive(Debug, Clone)]
pub struct MimeDecode<'a> {
    pub parsable_headers: bool,
    pub parsable_body: bool,
    pub ctnt_attachment: bool,
    pub headers: Option<Headers<'a>>,
}
```
I'm not very well experienced with Rust but I tried implementing the `Debug` trait for `Headers` and found out we cannot do that on entities imported from external crates.
My question is (if all my above deduction is correct), is there a reason we do not have `Debug` and `Clone` implemented for this struct? I just wish to learn and maybe restructure what I am writing if we do not have these traits for a reason.

Thank you very much for taking out time to read! :)

**Comment by staktrace** [comment]
Hi! So it is certainly possible to implement the `Debug` trait on `Headers` but I'm not sure that the implementation generated by `#[derive(Debug)]` would be very useful. This is because the `Headers` struct has things like `&[u8]` which is just an array of bytes and dumping that to the console wouldn't render things nicely. It would be better to manually implement `Debug` to print more usefully-formatted key/value pairs of headers. I'd accept a patch to do that, if you're interested in trying.

As for `Clone`, I don't think that makes as much sense. One of the key design features of the `mailparse` library is that it tries to avoid copying things, and `Clone` by definition copies things. The lifetime parameters on the various structs in `mailparse` are intended to make it usable in a zero-copy manner. I agree that implementing `Clone` wouldn't necessary affect how people use the library, but they would certainly be more inclined to just copy stuff around without thinking about the whole-program architecture. I hope that makes sense.

Feel free to ask more detailed questions if you have them; my answer above is a bit generic because I haven't spent much time on `mailparse` lately and it's paged out of my head.

**Comment by inashivb** [comment]
Thank you very much for the detailed explanation @staktrace ! I'll send a PR for Debug trait impl. This is all I needed to know.

**Comment by staktrace** [comment]
Fixed by #86.
